test/e2e/framework/job/wait.go

```diff
@@ -196,14 +214,18 @@ func WaitForJobFinish(ctx context.Context, c clientset.Interface, ns, jobName st
 
 // WaitForJobFinishWithTimeout uses c to wait for the Job jobName in namespace ns to finish (either Failed or Complete).
 func WaitForJobFinishWithTimeout(ctx context.Context, c clientset.Interface, ns, jobName string, timeout time.Duration) error {
-	return wait.PollUntilContextTimeout(ctx, framework.Poll, timeout, true, func(ctx context.Context) (bool, error) {
-		curr, err := c.BatchV1().Jobs(ns).Get(ctx, jobName, metav1.GetOptions{})
-		if err != nil {
-			return false, err
-		}
-
-		return isJobFinished(curr), nil
-	})
+	return framework.Gomega().
+		Eventually(ctx, framework.RetryNotFound(framework.GetObject(c.BatchV1().Jobs(ns).Get, jobName, metav1.GetOptions{}))).
+		WithPolling(framework.Poll).
+		WithTimeout(timeout).
+		Should(framework.MakeMatcher(func(job *batchv1.Job) (func() string, error) {
+			if isJobFinished(job) {
+				return nil, nil
+			}
+			return func() string {
+				return fmt.Sprintf("expected job to be finished, got instead:\n%s", format.Object(job, 1))
```

Comment:
```suggestion
				return fmt.Sprintf("expected job %q to be finished\n%s", klog.KObj(job), format.Object(job, 1))
```

Reply:
@kaisoz `format.Object(job, 1)` will print the job's metadata with the yaml format. We may not need to print the job name anymore.

Reply:
yeah, I get your point. This is just for consistency since the rest of the functions log the job name as well. I'd follow the current way or change them all (but the latter maybe in a different PR 🤷🏻 )